src/data_provider/src/packages/packageLinuxParserApk.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * Wazuh SYSINFO
+ * Copyright (C) 2015, Wazuh Inc.
+ * December 19, 2022.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include "json.hpp"
+#include <fstream>
+
+void getApkInfo(const std::string& fileName, std::function<void(nlohmann::json&)> callback)
+{
+    std::ifstream apkDb(fileName);
+    const size_t valuePos {2};
+    std::string line;
+    std::vector<std::string> data;
```

Comment:
If you know the maximum size of the vector, maybe you can predefine the size.

[how-to-set-initial-size-of-stdvector](https://stackoverflow.com/questions/11457571/how-to-set-initial-size-of-stdvector)

Reply:
I use an array to declare the fixed number of keys we are interested in.